File: #1-25 (C, Java, Python)/Python/Euler15.py
#Euler15 ~ Lattice paths
import timeit
start = timeit.default_timer()

#Lattice Paths of Cube N --> Central Binomial Coefficients: (2n)!/(n!)^2
cubeSize = 20

#Find the numerator value
factorialN = cubeSize*2
numerator = 1
for i in range(1,(factorialN+1)):
    numerator = numerator*i
    
denominator = 1
for j in range(1,(cubeSize+1)):
    denominator = denominator*j

denominator = denominator*denominator

solution = numerator/denominator

stop = timeit.default_timer()
print("Solution: ",solution,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")

File: #1-25 (C, Java, Python)/Java/Euler15.java
//Euler15 ~ Lattice paths
import java.util.concurrent.TimeUnit;
import java.math.BigInteger;

public class Euler15
{
    public static void main(String[] args) throws InterruptedException
	{
		long startTime = System.nanoTime();
		
		//Lattice Paths of Cube N --> Central Binomial Coefficients: (2n)!/(n!)^2
		int cubeSize = 20;
		
		//Find the numerator value
		int factorialN = cubeSize*2;
		BigInteger numerator = new BigInteger("1");
		BigInteger iBig = new BigInteger("1");
		for(int i = 1;i <= factorialN; i++)
		{
			iBig = BigInteger.valueOf(i);
			numerator = numerator.multiply(iBig);
		}
		
		//Find the denominator value
		BigInteger denominator = new BigInteger("1");
		BigInteger jBig = new BigInteger("1");
		for(int j = 1;j <= cubeSize; j++)
		{
			jBig = BigInteger.valueOf(j);
			denominator = denominator.multiply(jBig);
		}
		denominator = denominator.multiply(denominator);
		
		//Find path count
		BigInteger solution = numerator.divide(denominator);
		
		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + solution + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
    }
}